debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -66,13 +67,23 @@
      * Creates a Postgres connection using the supplied configuration.
      *
      * @param config {@link Configuration} instance, may not be null.
+     * @param provideTypeRegistry {@code true} if type registry should be created
      */
-    public PostgresConnection(Configuration config) {
+    public PostgresConnection(Configuration config, boolean provideTypeRegistry) {
         super(config, FACTORY, PostgresConnection::validateServerVersion, PostgresConnection::defaultSettings);
-        this.typeRegistry = new TypeRegistry(this);
+        this.typeRegistry = provideTypeRegistry ? new TypeRegistry(this) : null;
         databaseCharset = determineDatabaseCharset();
     }
 
+    /**
+     * Creates a Postgres connection using the supplied configuration.
+     *
+     * @param config {@link Configuration} instance, may not be null.
+     */
+    public PostgresConnection(Configuration config) {
+        this(config, false);
```

Comment:
Shouldn't it rather be true by default?

Reply:
That said, I don't quite get how it's meant to work to begin with. Could you comment in the test? Thx!

Reply:
ok, let me explain. Right now when you create a postgresconnection then in all instances it creates internal map of types which means additional queries and transactions. The map is needed for only one single specific connection. So I decided it makes sense to handle this connection spcefically and by default not to create the type registry for connections that does not need it.

Reply:
OK, gotcha. Could you put this as a source comment into a sensible location. Thx!

Reply:
Added, should be ready for review now.

Reply:
Thx!